NPE when refreshing sitemap

I just found this in my log of an openHAB 2.3 release version when doing (potentially invalid) changes to a sitemap file:

```
22:26:40.546 [INFO ] [del.core.internal.ModelRepositoryImpl] - Loading model 'default.sitemap'
22:26:40.561 [WARN ] [del.core.internal.ModelRepositoryImpl] - Configuration model 'default.sitemap' is either empty or cannot be parsed correctly!
22:26:40.562 [ERROR] [l.core.internal.folder.FolderObserver] - Error handling update of file '/Users/kai/Documents/dev/demo/conf/sitemaps/default.sitemap': null.
java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011) [?:?]
	at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006) [?:?]
	at org.eclipse.smarthome.model.sitemap.internal.SitemapProviderImpl.modelChanged(SitemapProviderImpl.java:106) [140:org.eclipse.smarthome.model.sitemap:0.10.0.oh230]
	at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.notifyListeners(ModelRepositoryImpl.java:301) [126:org.eclipse.smarthome.model.core:0.10.0.oh230]
	at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:123) [126:org.eclipse.smarthome.model.core:0.10.0.oh230]
	at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:227) [126:org.eclipse.smarthome.model.core:0.10.0.oh230]
	at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.processWatchEvent(FolderObserver.java:291) [126:org.eclipse.smarthome.model.core:0.10.0.oh230]
	at org.eclipse.smarthome.core.service.WatchQueueReader.run(WatchQueueReader.java:240) [101:org.eclipse.smarthome.core:0.10.0.oh230]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```

--- a/bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java
+++ b/bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java
@@ -107,7 +107,7 @@ public class SitemapProviderImpl implements SitemapProvider, ModelRepositoryChan
                 EObject sitemap = modelRepo.getModel(modelName);
                 // if the sitemap file is empty it will not be in the repo and thus there is no need to cache it here
                 if (sitemap instanceof Sitemap) {
-                    sitemapModelCache.put(modelName, (Sitemap) modelRepo.getModel(modelName));
+                    sitemapModelCache.put(modelName, (Sitemap) sitemap);
                 }
             }
         }

--- a/bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java
+++ b/bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java
@@ -19,6 +19,7 @@ import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.stream.Collectors;
 
 import org.apache.commons.lang.StringUtils;
+import org.eclipse.emf.ecore.EObject;
 import org.eclipse.smarthome.model.core.EventType;
 import org.eclipse.smarthome.model.core.ModelRepository;
 import org.eclipse.smarthome.model.core.ModelRepositoryChangeListener;
@@ -103,7 +104,11 @@ public class SitemapProviderImpl implements SitemapProvider, ModelRepositoryChan
             if (type == EventType.REMOVED) {
                 sitemapModelCache.remove(modelName);
             } else {
-                sitemapModelCache.put(modelName, (Sitemap) modelRepo.getModel(modelName));
+                EObject sitemap = modelRepo.getModel(modelName);
+                // if the sitemap file is empty it will not be in the repo and thus there is no need to cache it here
+                if (sitemap instanceof Sitemap) {
+                    sitemapModelCache.put(modelName, (Sitemap) modelRepo.getModel(modelName));
+                }
             }
         }
         for (ModelRepositoryChangeListener listener : modelChangeListeners) {

